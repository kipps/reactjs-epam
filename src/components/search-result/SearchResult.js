import React from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";

class SearchResult extends React.Component {
  render() {
    return (
      <p>
        <span><b>{this.props.posts.length}</b> movies found</span>
      </p>
    );
  }
}

SearchResult.propTypes = {
  posts: PropTypes.array
}

const mapStateToProps = state => {
  return {
    posts: state.moviesState.posts
  }
}

export default connect(mapStateToProps)(SearchResult)
