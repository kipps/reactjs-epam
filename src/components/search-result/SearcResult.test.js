import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import SearchResult from "./SearchResult";

configure({ adapter: new Adapter() });

describe("Search Result ", () => {
  it("renders default SearchResult", () =>
    expect(
      shallow(<SearchResult />)
        .find('span.red')
        .length
    ).toBe(1));

  it('renders SearchResult correctly', () => {
    const tree = renderer
      .create(<SearchResult />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});