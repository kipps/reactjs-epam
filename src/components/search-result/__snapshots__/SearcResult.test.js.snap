// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result  renders SearchResult correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="flex v-center space-between col"
    >
      <p>
        <span>
          <b>
            0
          </b>
           movies found
        </span>
      </p>
      <p>
        <span
          className="inline mr-8"
        >
          Sort by 
          <b>
            Release date
          </b>
        </span>
        |
        <span
          className="red inline ml-8"
        >
          Rating ___
        </span>
      </p>
    </div>
  </div>
</div>
`;
